javatests/google/registry/export/ExportConstantsTest.java

```diff
@@ -67,27 +69,65 @@ public void testBackupKinds_matchGoldenBackupKindsFile() throws Exception {
         .toList();
     ImmutableSet<String> actualKinds = ExportConstants.getBackupKinds();
-    String updateInstructions = String.format(
-        UPDATE_INSTRUCTIONS_TEMPLATE,
-        goldenBackupKindsResource.toString(),
-        Joiner.on('\n').join(actualKinds));
+    String updateInstructions = getUpdateInstructions("backed-up",
+        goldenBackupKindsResource.toString(), actualKinds );
     assertWithMessage(updateInstructions)
         .that(actualKinds)
         .containsExactlyElementsIn(goldenKinds)
         .inOrder();
   }
 
   @Test
+  public void testReportingKinds_matchGoldenReportingKindsFile() throws Exception {
+    URL goldenReportingKindsResource =
+        getResource(ExportConstantsTest.class, GOLDEN_REPORTING_KINDS_FILENAME);
+    final Pattern stripComments = Pattern.compile("\\s*#.*$");
+    List<String> goldenReportingKinds = FluentIterable
+        .from(Splitter.on('\n').split(
+            Resources.toString(goldenReportingKindsResource, UTF_8).trim()))
+        .transform(
+            new Function<String, String>() {
+              @Override @Nullable public String apply(@Nullable String line) {
+                return stripComments.matcher(line).replaceFirst("");
+              }})
+        .toList();
+    ImmutableSet<String> actualKinds = ExportConstants.getReportingKinds();
+    String updateInstructions = getUpdateInstructions("reporting",
+        goldenReportingKindsResource.toString(), actualKinds );
+    assertWithMessage(updateInstructions)
+        .that(actualKinds)
+        .containsExactlyElementsIn(goldenReportingKinds)
+        .inOrder();
+  }
+
+  @Test
   public void testReportingKinds_areSubsetOfBackupKinds() throws Exception {
     assertThat(ExportConstants.getBackupKinds()).containsAllIn(ExportConstants.getReportingKinds());
   }
 
-  @Test
-  public void testReportingEntityClasses_areAllBaseEntityClasses() throws Exception {
-    for (Class<? extends ImmutableObject> clazz : ExportConstants.REPORTING_ENTITY_CLASSES) {
-      assertThat(clazz.isAnnotationPresent(Entity.class))
-          .named(String.format("class %s is an @Entity", clazz.getSimpleName()))
-          .isTrue();
-    }
+  @Test 
+  public void testReportingKinds_areSubsetOfEntityClasses() throws Exception {  
+    ImmutableSet<String> entityClasses =  FluentIterable.from(EntityClasses.ALL_CLASSES) 
+        .filter(hasAnnotation(Entity.class)) 
+        .transform(CLASS_TO_KIND_FUNCTION) 
+        .toSortedSet(Ordering.natural()); 
+    assertThat(entityClasses).containsAllIn(ExportConstants.getReportingKinds())
+   }
+
+  /**
+   * Helper method to get update instructions
+   *
+   * @param name - type of entity
+   * @param resource - Resource file contents
+   * @param actualKinds - data from ExportConstants
+   * @return String of update instructions
+   */
+  private String getUpdateInstructions(String name, String resource,
+                                       ImmutableSet<String> actualKinds){
```

Comment:
Please run google-java-format over your code - e.g. there should be a space between ) and {, and the parameters should be indented +4 after being wrapped, not aligned with the open-paren.